internal/helm/helmadapter/gorm_repo_store.go

```diff
@@ -0,0 +1,146 @@
+	return helmRepoStore{
+		db:     db,
+		logger: logger,
+	}
+}
+
+func (h helmRepoStore) Delete(_ context.Context, organizationID uint, repository helm.Repository) error {
+	model := toModel(repository)
+	model.OrganizationID = organizationID
+
+	// find soft-deleted records if any
+	if err := h.db.Unscoped().Where(model).First(&model).Error; err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return nil
+		}
+		return errors.WrapIf(err, "failed to load helm repository record")
+	}
+
+	// delete the record permanently in order for the unique constraint to be workings
+	if err := h.db.Unscoped().Delete(model).Error; err != nil {
+		return errors.WrapIf(err, "failed to delete repository record")
+	}
+
+	h.logger.Debug("deleted helm repository record",
+		map[string]interface{}{"organisationID": organizationID, "repoName": repository.Name})
+
+	return nil
+}
+
+func (h helmRepoStore) List(_ context.Context, organizationID uint) ([]helm.Repository, error) {
+	var repoModels []repositoryModel
+	if err := h.db.Where("organization_id = ?", organizationID).Find(&repoModels).Error; err != nil {
+		return nil, errors.WrapIf(err, "failed to list helm repositories")
+	}
+
+	repos := make([]helm.Repository, 0, len(repoModels))
+	for _, model := range repoModels {
+		repos = append(repos, toDomain(model))
+	}
+
+	h.logger.Debug(
+		"retrieved helm repository records",
+		map[string]interface{}{
+			"organisationID": organizationID,
+			"repositories #": len(repos)})
+
+	return repos, nil
+}
+
+func (h helmRepoStore) Create(_ context.Context, organizationID uint, repository helm.Repository) error {
+	repoModel := toModel(repository)
+	repoModel.OrganizationID = organizationID
+
+	if err := h.db.Create(&repoModel).Error; err != nil {
+		return errors.WrapIf(err, "failed to persist the helm repository")
+	}
+
+	h.logger.Debug(
+		"persisted new helm repository record",
+		map[string]interface{}{
+			"organisationID": organizationID,
+			"repoName":       repository.Name})
+
+	return nil
+}
+
+func (h helmRepoStore) Get(_ context.Context, organizationID uint, repository helm.Repository) (helm.Repository, error) {
+	repoModel := toModel(repository)
+	repoModel.OrganizationID = organizationID
+
+	if err := h.db.Where(&repoModel).First(&repoModel).Error; err != nil {
+		return helm.Repository{}, errors.WrapIf(err, "failed to get helm repository")
+	}
+	h.logger.Debug("retrieved helm repository record",
+		map[string]interface{}{"organisationID": organizationID, "repoName": repository.Name})
+
+	return toDomain(repoModel), nil
+}
+
+// toDomain transforms a gorm model to a domain struct
+func toDomain(model repositoryModel) helm.Repository {
+	return helm.Repository{
+		Name:             model.Name,
+		URL:              model.URL,
+		PasswordSecretID: model.PasswordSecretID,
+		TlsSecretID:      model.TlsSecretID,
+	}
+}
+
+//toModel transforms a domain struct to gorm model representation
+func toModel(repository helm.Repository) repositoryModel {
```

Comment:
Why not pass the org ID also, instead of always setting it afterwards?
```suggestion
func toModel(organizationID uint, repository helm.Repository) repositoryModel {
```